### PR TITLE
feat: set in-app route from screen tracking

### PR DESCRIPTION
**Pull request by levibostian**
Complete each step to get your pull request merged in. [Learn more about the workflow this project uses](https://github.com/customerio/customerio-ios/blob/develop/docs/dev-notes/GIT-WORKFLOW.md). 
- [x] [Assign members of your team](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/requesting-a-pull-request-review) to review the pull request. 
- [x] Wait for pull request status checks to complete. If there are problems, fix them until you see that [all status checks are passing](https://external-content.duckduckgo.com/iu/?u=https%3A%2F%2Fsymfony.com%2Fdoc%2F4.3%2F_images%2Fdocs-pull-request-symfonycloud.png&f=1&nofb=1). 
- [ ] Wait until the pull request has been reviewed *and approved* by a teammate
- [ ] After pull request is approved, and you determine it's ready **add the label "Ready to merge"** to the pull request. A bot will *squash and merge* the pull request for you after the label is added.

### CODE REVIEW

File: Sources/MessagingPush/hooks/ModuleHook.swift [hunk 1]
```diff
@@ -28,4 +28,8 @@ class MessagingPushModuleHookProvider: ModuleHookProvider {
     var deviceAttributesHook: DeviceAttributesHook? {
         MessagingPushImplementation(siteId: siteId)
     }
+
+    var screenTrackingHook: ScreenTrackingHook? {
```
Comment (jatinn) [inline Sources/MessagingPush/hooks/ModuleHook.swift:32]: is this needed even though it returns nil?

File: Sources/Tracking/CustomerIO.swift [hunk 2]
```diff
@@ -360,7 +360,7 @@ public class CustomerIO: CustomerIOInstance {
         implementation?.screen(name: name, data: data)
     }
 
-    public func automaticScreenView(
+    internal func automaticScreenView(
```
Comment (jatinn) [inline Sources/Tracking/CustomerIO.swift:363]: is this going to be a breaking change since making public functions internal now?

**Review by jatinn — APPROVED** [review]

File: Sources/Tracking/CustomerIO.swift (hunk 2 above)
Comment (levibostian) [reply Sources/Tracking/CustomerIO.swift:363]: This function is undocumented and was not meant for customers to use. The function currently being marked as `public` is a bug IMO. So, no. I do not plan on introducing a breaking change.

File: Sources/MessagingPush/hooks/ModuleHook.swift (hunk 1 above)
Comment (levibostian) [reply Sources/MessagingPush/hooks/ModuleHook.swift:32]: In order for the class to satisfy the requirements of the protocol, yes, this property is required in the code for it to compile. 

I could fix this by introducing a class that gets extended... 🤔

**Pull request merged**